pkg/generators/rules/names_match.go

```diff
@@ -78,7 +78,13 @@ func (n *NamesMatch) Validate(t *types.Type) ([]string, error) {
 	case types.Struct:
 		for _, m := range t.Members {
 			goName := m.Name
-			jsonTag := reflect.StructTag(m.Tags).Get("json")
+			jsonTag, ok := reflect.StructTag(m.Tags).Lookup("json")
+			// Distinguish empty JSON tag and missing JSON tag. Empty JSON tag / name is
+			// allowed (in JSON name blacklist) but missing JSON tag is invalid.
```

Comment:
Thanks, great. I wonder if we should support e.g. "-" as a json name, indicating that a custom marshaller will be used? "" as a name is used for inlining, we should keep that distinct.

Reply:
Yes, currently we support "", "-" and "metadata" as valid json names (see `jsonNameBlacklist`). 

I'm thinking if we want be more specific and evaluate based on json tag instead of json name (`jsonName := strings.Split(jsonTag, ",")[0]`). E.g.:
```
// From https://golang.org/pkg/encoding/json/#Marshal

/* Case 1 */
// Field is ignored by this package.
Field int `json:"-"`					// This is valid

// Field appears in JSON as key "-".
Field int `json:"-,"`					// This should be wrong for the API rule, but currently it's allowed

/* Case 2 */
// Field appears in JSON as key "Field" (the default), but
// the field is skipped if empty.
// Note the leading comma.
Field int `json:",omitempty"`				// This is valid

metav1.TypeMeta `json:",inline"`			// This is valid in Kubernetes

Field int `json:""`					// Is this one valid?

/* Case 3 */
metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`		// This is valid in Kubernetes

Field int `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`			// Should we catch this case?
```

Reply:
Updated to make 
```
Field int `json:"-,"`
``` 
invalid